File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter04/C/solutionsOfSecondDegreeEquation.c
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 04.-Flujo de control I: Estructuras Selectivas
    Problema 13.-Pseudocodigo que nos permita calcular las soluciones de una ecuacion de segundo grado, incluyendo
    los valores imaginarios.
    ejemplo 4.17, Pagina 147
*/
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

void main(){
    float a=0, b=0, c=0, d=0, x1=0, x2=0, r=0, i=0;
    printf("Programa que nos permita calcular las soluciones de una ecuacion de segundo grado, incluyendo los valores imaginarios.");

    printf("\n\nIngrese los coeficientes: ");
    printf("\nIngrese el valor de A: ");
    scanf("%f", &a);
    printf("Ingrese el valor de B: ");
    scanf("%f", &b);
    printf("Ingrese el valor de C: ");
    scanf("%f", &c);

    if(a==0){
        printf("La ecuacion no es de segundo grado.\n");
    }else{
        d=b*b-4*a*c;

        if(d==0){
            x1= -b/(2+a);
            x2=x1;
            printf("Los valores son para:\nx1=%1.2f\nx2=%1.2f", x1, x2);

        }else if(d>0){
            x1=(-b +sqrt(d)) / (2*a);
            x2=(-b -sqrt(d)) / (2*a);
            printf("\nLos valores son para:\nx1=%1.2f\nx2=%1.2f", x1, x2);
        }else{
            r=(-b)/(2*a);
            int dAbs=(int)(d);
            int absNum=abs(dAbs);
            i=sqrt(absNum) /(2*a);
            printf("\n\nLos valores son para:\nr=%1.2f + %1.2fi", r, i);
            printf("\nLos valores son para:\nr=%1.2f - %1.2fi", r, i);
        }
    }
    printf("\n\n");
}

File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter04/JAVASCRIPT/js/solutionsOfSecondDegreeEquation.js
/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 04.-Flujo de control I: Estructuras Selectivas
    Problema 13.-Pseudocodigo que nos permita calcular las soluciones de una ecuacion de segundo grado, incluyendo
    los valores imaginarios.
    ejemplo 4.17, Pagina 147
*/
function calculate(){
    var a=0, b=0, c=0, d=0, x1=0, x2=0, r=0, i=0;
    var varSpan=document.createElement("span");
        varSpan.setAttribute("id", "idSpan");           
             
    if(document.getElementById("idBtnCalculate").value=="Calcular"){                     
        a=parseFloat(document.getElementById("idA").value);
        b=parseFloat(document.getElementById("idB").value);
        c=parseFloat(document.getElementById("idC").value);

        if(a==0){
            varSpan.innerHTML="La ecuacion no es de segundo grado.<br>";
        }else{
            d=b*b-4*a*c;

            if(d==0){
                x1= -b/(2+a);
                x2=x1;
                varSpan.innerHTML="Los valores son para:<br>x1="+x1+"<br>x2="+x2+"<br>";

            }else if(d>0){
                x1=(-b +Math.sqrt(d)) / (2*a);
                x2=(-b -Math.sqrt(d)) / (2*a);
                varSpan.innerHTML="Los valores son para:<br>x1="+x1+"<br>x2="+x2+"<br>";
            }else{
                r=(-b)/(2*a);
                //var dAbs=Math.abs(dAbs);
                i=Math.sqrt(Math.abs(d)) /(2*a);
                varSpan.innerHTML+="<br>Los valores son para:<br>r="+r+" + "+i+"i<br>";
                varSpan.innerHTML+="<br>Los valores son para:<br>r="+r+" - "+i+"i<br>";
            }
        }


        document.getElementById("idSection").appendChild(varSpan);    
        document.getElementById("idBtnCalculate").value="Limpiar";
    }else{
        document.getElementById("idA").value="";
        document.getElementById("idB").value="";
        document.getElementById("idC").value="";
        document.getElementById("idSpan").remove();
        
        document.getElementById("idBtnCalculate").value="Calcular";
    }
}  